compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2484,13 +2486,28 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       def newMethod(owner: Symbol, name: String, tpe: TypeRepr): Symbol =
         newMethod(owner, name, tpe, Flags.EmptyFlags, noSymbol)
       def newMethod(owner: Symbol, name: String, tpe: TypeRepr, flags: Flags, privateWithin: Symbol): Symbol =
+        import Flags.*
+        // TODO: missing AbsOverride
+        checkValidFlags(flags.toTermFlags, Private | Protected | Override | Deferred | Final | Method | Implicit | Given | Local | JavaStatic) // Synthetic | ExtensionMethod | Exported | Erased | Infix | Invisible
```

Comment:
It would be nicer to create a val for each of these long lists of flags. Also couldn't this be checked in -Ycheck?

Reply:
The idea of the `-Xcheck-macros` flag is to help users find the location in their implementation where the bug started. Additionally, waiting until `-Ycheck` can result in other derived failures or crashes before we reach the checks.